### PR TITLE
Feature/lu 4538

### PR DESCRIPTION
**Pull request by sivagudur**

### CODE REVIEW

File: src/main/java/uk/gov/ons/collection/service/BatchDataIngest.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+            log.error("Invalid JSON from contributor exists query response " + e);
+            throw new InvalidJsonException("Invalid JSON from contributor exists query response: " + individualObjectJsonStr, e);
+        }
+
+    }
+
+
+    public boolean isContributorEmpty(String referenceExistsResponse) throws InvalidJsonException {
+
+        JSONObject referenceExistsObject;
```
Comment (dllmorgan) [inline src/main/java/uk/gov/ons/collection/service/BatchDataIngest.java]: JSONObject referenceExistsObject;  can be moved to line 117 as -> var referenceExistsObject
as it's not used outside of the try {} scope

File: src/main/java/uk/gov/ons/collection/service/BatchDataIngest.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+                variables.put("period", period);
+                variables.put("survey", survey);
+                outcomeObject.put("reference", reference);
+                outcomeObject.put("period", period);
+                outcomeObject.put("survey", survey);
+                referenceExistsResponse = qlService
+                            .qlSearch(new BatchDataQuery(variables).buildCheckReferenceExistsQuery());
+                log.info("Reference exists response:" + referenceExistsResponse);
+                if (isContributorEmpty(referenceExistsResponse)) {
+                    StringBuilder sbContribError = new StringBuilder("Contributor doesn't exist in database ");
```
[the remark this answers is not in the document]
Comment (sivagudur) [reply src/main/java/uk/gov/ons/collection/service/BatchDataIngest.java]: Done and added a private method buildOutcomeJson

File: src/main/java/uk/gov/ons/collection/service/BatchDataIngest.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+    }
+
+    public BatchDataIngest(String batchResponses, GraphQlService qlGraphService) throws InvalidJsonException {
+
+        JSONObject inputJson;
+        try {
+            inputJson = new JSONObject(batchResponses);
+            referenceArray = inputJson.getJSONArray("batch_data");
+            qlService = qlGraphService;
```
[the remark this answers is not in the document]
Comment (sivagudur) [reply src/main/java/uk/gov/ons/collection/service/BatchDataIngest.java]: Done

File: src/main/java/uk/gov/ons/collection/entity/BatchDataQuery.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    public BatchDataQuery(Map<String, String> variables) throws InvalidJsonException {
+        try {
+            inputKey = new JSONObject(variables);
+        } catch (JSONException e) {
+            throw new InvalidJsonException("Can't build Query for checking if reference/period/survey exists for Batch ingest of data: " + inputKey, e);
+        }
+    }
+
+    public String buildCheckReferenceExistsQuery() {
```
[the remark this answers is not in the document]
Comment (sivagudur) [reply src/main/java/uk/gov/ons/collection/entity/BatchDataQuery.java]: Added a InvalidJsonException to calling method buildCheckReferenceExistsQuery()

File: src/main/java/uk/gov/ons/collection/service/BatchDataIngest.java (hunk 1 above)
Comment (sivagudur) [reply src/main/java/uk/gov/ons/collection/service/BatchDataIngest.java]: Done

**Pull request merged**